rpc/backend/backend.go

```diff
@@ -104,7 +104,7 @@ func (b *EthermintBackend) GetBlockByHash(hash common.Hash, fullTx bool) (map[st
 
 	resBlock, err := b.clientCtx.Client.Block(&out.Number)
 	if err != nil {
-		return nil, err
+		return nil, nil
```

Comment:
why are we not returning an error? seems like this could cause a segfault in the next line if the error isn't returned

Reply:
the function `GetBlockByHash` was only used by rpc interface `PublicEthereumAPI::GetBlockByHash`, in case of query block by hash, if can not found target block found by hash, the calling result should be `null`, if return error here, the caller will receive an error and not `null`, some of dev tools in ecosystem of ethereum will throw an error, certainly, the better way is return the error to caller, but i think we need to keep sync with call result of ethereum rpc. 
so, the root cause is ethereum rpc will return nil here even in case of error

Reply:
okay, that makes sense. if it improves tooling then that is fine